preprocess.py

```diff
@@ -113,14 +109,91 @@ def preprocess(args):
         )
-        test_dataset = ShakeSpeare(train=False)
+        testset = ShakeSpeare(train=False)
         test_loader = torch.utils.data.DataLoader(
-            test_dataset, batch_size=args.test_batch_size, shuffle=False, **kwargs
+            testset, batch_size=args.test_batch_size, shuffle=False, **kwargs
         )
     return device, train_loader, test_loader
+
+
+def preprocess_cezo_fl(args):
+    if args.dataset == "mnist":
+        device, kwargs = use_device(args)
+        transform = transforms.Compose(
+            [transforms.ToTensor(), transforms.Normalize((0.1307,), (0.3081,))]
+        )
+        trainset = torchvision.datasets.MNIST(
+            root="./data", train=True, download=True, transform=transform
+        )
+        testset = torchvision.datasets.MNIST(
+            root="./data", train=False, download=True, transform=transform
+        )
+        test_loader = torch.utils.data.DataLoader(
+            testset, batch_size=args.test_batch_size, **kwargs
+        )
+    elif args.dataset == "cifar10":
+        device, kwargs = use_device(args)
+        transform_train = transforms.Compose(
+            [
+                transforms.RandomCrop(32, padding=4),
+                transforms.RandomHorizontalFlip(),
+                transforms.ToTensor(),
+                transforms.Normalize((0.4914, 0.4822, 0.4465), (0.2023, 0.1994, 0.2010)),
+            ]
+        )
+        trainset = torchvision.datasets.CIFAR10(
+            root="./data", train=True, download=True, transform=transform_train
+        )
+        transform_test = transforms.Compose(
+            [
+                transforms.ToTensor(),
+                transforms.Normalize((0.4914, 0.4822, 0.4465), (0.2023, 0.1994, 0.2010)),
+            ]
+        )
+        testset = torchvision.datasets.CIFAR10(
+            root="./data", train=False, download=True, transform=transform_test
+        )
+        test_loader = torch.utils.data.DataLoader(
+            testset, batch_size=args.test_batch_size, **kwargs
+        )
+    elif args.dataset == "fashion":
+        device, kwargs = use_device(args)
+        transform = transforms.Compose(
+            [transforms.ToTensor(), transforms.Normalize((0.5,), (0.5,))]
+        )
+        trainset = torchvision.datasets.FashionMNIST(
+            root="./data", train=True, download=True, transform=transform
+        )
+        testset = torchvision.datasets.FashionMNIST(
+            root="./data", train=False, download=True, transform=transform
+        )
+        test_loader = torch.utils.data.DataLoader(
+            testset, batch_size=args.test_batch_size, **kwargs
+        )
+    elif args.dataset == "shakespeare":
+        device, kwargs = use_device(args)
+        trainset = ShakeSpeare(train=True)
+        testset = ShakeSpeare(train=False)
+        test_loader = torch.utils.data.DataLoader(
+            testset, batch_size=args.test_batch_size, shuffle=False, **kwargs
+        )
```

Comment:
why not we reuse `preprocess` function here?

Reply:
Original preprocess loads 1 test dataset and build 1 dataloader using that dataset. 
However, we need to split that test dataset into num_clients subsets, then build num_clients dataloader. 

Original preprocess return is device: str, train_loader: DataLoader, test_loader: DataLoader
New preprocess_cezo_fl return is device: str, train_loaders: list[DataLoader], test_loader: DataLoader.

return types are different, so I decided to split them into 2 different functions